### PR TITLE
Setup composer scripts and a few debug packages

### PR DESCRIPTION
**Pull request by WindOfRussia**
### Summary

Continue basic laravel repo setup by adding debug packages and convenient composer scripts. 

### Test plan (required)
- various composer scripts run without errors
- composer fresh runs without errors

### Closing issues
List all issues this PR resolves.
- closes CSU-Booking-Platform#26

### Pre-Merge checklist 
Complete these items before requesting reviews
- [x] Static Analysis Rules Validated (local IDE only, no standard implemented so far)
- [x] All commits follow naming conventions

### CODE REVIEW

**Review by alexstojda — CHANGES_REQUESTED** [review]
Please rebase with the latest from main :)

**Comment by WindOfRussia** [comment]
Rebased from main repo

**Review by alexstojda — CHANGES_REQUESTED** [review]
You have a commit that does not follow naming conventions: 
`Setup additional packages - prequel & terminal` -> 709aa94db6ddd23916dd960828dfb909c2068878

**Comment by alexstojda** [comment]
@WindOfRussia  I see you committed the folder `public/vendor`, which only contains generated code. Do we need this in VCS or can this be generated as part of the deployment process?

**Review by Alex-Hayeur — DISMISSED** [review]
This is a small comment so I'll leave an approval for after the changes requested by Stojda are addressed, since this is related:

As he mentioned with a lot of the code in `public/vendor`, there's a lot of generated code from `resources/lang` to manage other languages. Does this also need to be committed or can it be added to the gitignore? We won't need the vast majority of these languages in our releases.

**Comment by WindOfRussia** [comment]
> @WindOfRussia I see you committed the folder `public/vendor`, which only contains generated code. Do we need this in VCS or can this be generated as part of the deployment process?

Any vendor assets that are published through artisan should be left in public. They don't get generated by the same process as the rest of our frontend and rarely change (only during package updates).

**Comment by WindOfRussia** [comment]
> This is a small comment so I'll leave an approval for after the changes requested by Stojda are addressed, since this is related:
> 
> As he mentioned with a lot of the code in `public/vendor`, there's a lot of generated code from `resources/lang` to manage other languages. Does this also need to be committed or can it be added to the gitignore? We won't need the vast majority of these languages in our releases.

Same as the last comment. Its safe to leave all the vendor generated files in both `public/vendor` and `resources/vendor` as this ensures that the ui of those packages works consistently for everyone even if we never modify them or end up needing to change the extra language configurations.

**Review by Alex-Hayeur — DISMISSED** [review]
Passing my approval after Andrew answered our questions.

File: public/vendor/prequel/mix-manifest.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+    "/app.js": "/app.js",
+    "/home/aether/Documents/Projects/HobbySoftware/laravel-sequel/packages/public/vendor/prequel/app.js": "/home/aether/Documents/Projects/HobbySoftware/laravel-sequel/packages/public/vendor/prequel/app.js"
```
Comment (danseremet) [inline public/vendor/prequel/mix-manifest.json:3]: Is this line specific to your machine?

**Review by danseremet — APPROVED** [review]
Looks good to me!

**Review by alexstojda — CHANGES_REQUESTED** [review]
I'm still not sold on including the vendor folders in the repo. I do understand your point about ensuring it works for everyone, but I think that assurance should be moved towards making sure everyone can generate these files. 

Case and point, none of these vendor assets are needed in the prod environment since these tools are not deployed in prod, so they would need to be deleted from the app when deploying. I don't think its best practice to be deleting things from the cloned repo as part of the deployment process. 

Also, artisan already has a command for exactly this purpose: `php artisan vendor:publish --tag=push [--force]`  https://laravel.com/docs/8.x/packages#public-assets

File: public/vendor/prequel/mix-manifest.json (hunk 1 above)
Comment (alexstojda) [reply public/vendor/prequel/mix-manifest.json:3]: Another reason these need to be `generated` and not stored in VCS :) 

@WindOfRussia

**Comment by WindOfRussia** [comment]
> I'm still not sold on including the vendor folders in the repo. I do understand your point about ensuring it works for everyone, but I think that assurance should be moved towards making sure everyone can generate these files.
> 
> Case and point, none of these vendor assets are needed in the prod environment since these tools are not deployed in prod, so they would need to be deleted from the app when deploying. I don't think its best practice to be deleting things from the cloned repo as part of the deployment process.
> 
> Also, artisan already has a command for exactly the purpose of generating these assets: `php artisan vendor:publish --tag=public [--force]` https://laravel.com/docs/8.x/packages#public-assets so we can just include it in our composer scripts & readme.

no problem, added public/vendor to the gitignore. The artisan vendor:publish command isn't the best example here since the public tag they give is just an example. Most packages come with their own tags and sometimes they cover more than just assets (models, config files, etc.) I've grouped these under a new composer script called `vendor-publish` which covers updating the assets for telescope, prequel and terminal packages.

File: public/vendor/prequel/mix-manifest.json (hunk 1 above)
Comment (WindOfRussia) [reply public/vendor/prequel/mix-manifest.json]: ouf, that is a line specific to the package developer's machine.

Comment (WindOfRussia) [reply public/vendor/prequel/mix-manifest.json]: it is also a mix manifest that is never used by anyone than the package developer so not critical but definitely useless to keep in vcs.

File: composer.json [hunk 2]
```diff
@@ -51,11 +60,77 @@
...
+            "@php artisan ide-helper:models -W"
+        ],
+        "ide:full": [
+            "@ide",
+            "@ide:models"
+        ],
+        "vendor-publish": [
+            "@php artisan telescope:publish --ansi",
+            "@php artisan prequel:update --ansi",
+            "@php artisan vendor:publish --tag=terminal --ansi"
```
Comment (alexstojda) [inline composer.json:133]: Could you please add documentation for all these commands to the README.md

File: resources/lang/vendor/prequel/de/lang.php [hunk 3]
```diff
@@ -0,0 +1,164 @@
+<?php
```
Comment (alexstojda) [inline resources/lang/vendor/prequel/de/lang.php:1]: Since we are going with the convention of generating vendor files & not storing them, these should be removed from VCS as well.

**Review by alexstojda — DISMISSED** [review]
As was done for `vendor/public` please remove the `resources/lang/vendor` folder as well, and add the relevant `vendor:publish` commands to the composer script. 

Also see above comment regarding documentation

**Comment by WindOfRussia** [comment]
TLDR: after talking to @alexstojda, we decided that the packages from the second commit (prequel, terminal) aren't necessary for dev environment and have equivalents both in codespaces and in standard local setups.

File: resources/lang/vendor/prequel/de/lang.php (hunk 3 above)
Comment (WindOfRussia) [reply resources/lang/vendor/prequel/de/lang.php]: Leaving this as a general note since we decided to remove prequel:
vendor subfolders in the `resources` folder are not for generated files, they are considered as override files for your composer packages. As such it would be bad practice to gitignore them. Their intended use is for modifying css, blade, vue, etc. files that are package developers define as default template or even some laravel components (like the mail & notification assets)

Typically you would only publish these when you explicitly intend to override those files, however the developer of the prequel package has decided that to publish these files during package installation.

File: public/vendor/telescope/app-dark.css [hunk 4]
```diff
@@ -0,0 +1,8 @@
+.vjs-tree{font-family:Monaco,Menlo,Consolas,Bitstream Vera Sans Mono,monospace!important}.vjs-tree .vjs-tree__content{border-left:1px dotted hsla(0,0%,80%,.28)!important}.vjs-tree .vjs-tree__node{cursor:pointer}.vjs-tree .vjs-tree__node:hover{color:#20a0ff}.vjs-tree .vjs-checkbox{position:absolute;left:-30px}.vjs-tree .vjs-value__boolean,.vjs-tree .vjs-value__null,.vjs-tree .vjs-value__number{color:#a291f5!important}.vjs-tree .vjs-value__string{color:#dacb4d!important}.hljs-addition,.hljs-keyword,.hljs-selector-tag{color:#8bd72f}.hljs-doctag,.hljs-meta .hljs-meta-string,.hljs-regexp,.hljs-string{color:#dacb4d}.hljs-literal,.hljs-number{color:#a291f5!important}
```
Comment (alexstojda) [inline public/vendor/telescope/app-dark.css:1]: As per earlier reviews, please .gitignore `public/vendor`

**Review by alexstojda — DISMISSED** [review]

File: public/vendor/telescope/app-dark.css (hunk 4 above)
Comment (WindOfRussia) [reply public/vendor/telescope/app-dark.css]: it be ignored. 10/10 for git resets but forgetting to delete existing files.

**Review by alexstojda — APPROVED** [review]
41 files -> 20, Much better :D

**Review by Alex-Hayeur — APPROVED** [review]
I kept accidentally approving when changes had already been requested causing the approval to be removed.
Passing approval on now.
